dask/dataframe/rolling.py

```diff
@@ -391,20 +392,34 @@ def quantile(self, quantile):
         return self._call_method("quantile", quantile)
 
     @derived_from(pd_Rolling)
-    def apply(self, func, args=(), kwargs={}, **kwds):
-        # TODO: In a future version of pandas this will change to
-        # raw=False. Think about inspecting the function signature and setting
-        # to that?
-        if PANDAS_VERSION >= "0.23.0":
-            kwds.setdefault("raw", None)
-        else:
-            if kwargs:
-                msg = (
-                    "Invalid argument to 'apply'. Keyword arguments "
-                    "should be given as a dict to the 'kwargs' argument. "
-                )
-                raise TypeError(msg)
-        return self._call_method("apply", func, args=args, kwargs=kwargs, **kwds)
+    def apply(
+        self,
+        func,
+        raw=None,
+        engine="cython",
+        engine_kwargs=None,
+        args=None,
+        kwargs=None,
+    ):
+        compat_kwargs = {}
+        kwargs = kwargs or {}
+        args = args or ()
+        meta = self.obj._meta.rolling(0)
+        if has_keyword(meta, "engine"):
+            # PANDAS_GT_100
+            compat_kwargs = dict(engine=engine, engine_kwargs=engine_kwargs)
+        elif engine != "cython" or engine_kwargs is not None:
+            raise NotImplementedError(
+                f"Specifying the engine requires pandas>=1.0.0. Version '{PANDAS_VERSION}' installed."
+            )
+
+        if raw is None and PANDAS_GT_100:
+            # pandas changed the default
+            raw = False
```

Comment:
This is another situation where it can possibly change the behavior of cuDF partitions based on the installed Pandas version. Is there any way to only do this if the partition is of type Pandas?

Reply:
I suppose we would need to inspect `inspect.signature(meta.apply)` and fish out the default argument?